components/psst/browser/core/brave_psst_permission_context.cc

```diff
@@ -0,0 +1,147 @@
+constexpr char kConsentStatusPermissionKey[] = "consent_status";
+constexpr char kScriptVersionPermissionKey[] = "script_version";
+constexpr char kUrlsToSkipPermissionKey[] = "urls_to_skip";
+
+base::Value::List VectorToList(const std::vector<std::string>& values) {
+  base::Value::List list;
+  list.reserve(values.size());
+  for (const auto& value : values) {
+    list.Append(value);
+  }
+  return list;
+}
+
+base::Value::Dict CreatePsstPermissionObject(
+    PsstPermissionInfo psst_permission_info) {
+  base::Value::Dict object;
+  object.Set(kUserIdPermissionKey, psst_permission_info.user_id);
+  object.Set(kConsentStatusPermissionKey,
+             ToString(psst_permission_info.consent_status));
+  object.Set(kScriptVersionPermissionKey, psst_permission_info.script_version);
+  object.Set(kUrlsToSkipPermissionKey,
+             VectorToList(psst_permission_info.urls_to_skip));
+  return object;
+}
+
+bool IsAllowedToProcess(const url::Origin& origin, std::string_view user_id) {
+  return origin.scheme() == url::kHttpsScheme && !user_id.empty();
+}
+
+}  // namespace
+
+BravePsstPermissionContext::BravePsstPermissionContext(
+    HostContentSettingsMap* host_content_settings_map)
+    : ObjectPermissionContextBase(ContentSettingsType::BRAVE_PSST,
+                                  host_content_settings_map) {}
+BravePsstPermissionContext::~BravePsstPermissionContext() = default;
+
+std::string BravePsstPermissionContext::GetKeyForObject(
+    const base::Value::Dict& object) {
+  const auto* user_id = object.FindString(kUserIdPermissionKey);
+  CHECK(user_id);
+  return *user_id;
+}
+
+bool BravePsstPermissionContext::IsValidObject(
+    const base::Value::Dict& object) {
+  const auto* user_id = object.FindString(kUserIdPermissionKey);
+  const auto* consent_status = object.FindString(kConsentStatusPermissionKey);
+  const auto script_version = object.FindInt(kScriptVersionPermissionKey);
+  const auto* urls_to_skip = object.FindList(kUrlsToSkipPermissionKey);
+
+  return user_id && !user_id->empty() && consent_status &&
+         !consent_status->empty() && script_version.has_value() && urls_to_skip;
+}
+
+std::u16string BravePsstPermissionContext::GetObjectDisplayName(
+    const base::Value::Dict& object) {
+  return base::UTF8ToUTF16(GetKeyForObject(object));
+}
+
+void BravePsstPermissionContext::CreateOrUpdate(
+    const url::Origin& origin,
+    ConsentStatus consent_status,
+    int script_version,
+    std::string_view user_id,
+    const base::Value::List& urls_to_skip) {
+  auto permission_info = PsstPermissionInfo::FromValue(
+      base::Value::Dict()
+          .Set(kUserIdPermissionKey, user_id)
+          .Set(kConsentStatusPermissionKey, static_cast<int>(consent_status))
+          .Set(kScriptVersionPermissionKey, script_version)
+          .Set(kUrlsToSkipPermissionKey, urls_to_skip.Clone()));
+  if (!permission_info) {
+    return;
+  }
+
+  CreateOrUpdate(origin, permission_info->Clone());
```

Comment:
please don't Clone/copy values and strings everywhere. Use `std::move()` accordingly.